src/Decompiler/Analysis/CallApplicationBuilder.cs

```diff
@@ -198,5 +213,24 @@ private Statement InsertStatementBefore(Instruction instr, Statement stmBefore)
             var linAddr = stmBefore.LinearAddress;
             return block.Statements.Insert(iPos, linAddr, instr);
         }
+
+        private Identifier EnsureRegister(RegisterStorage reg)
+        {
+            var id = ssaCaller.Procedure.Frame.EnsureRegister(reg);
+            var sid = EnsureSsaIdentifier(id);
+            sid.Uses.Add(stmCall);
+            return sid.Identifier;
+        }
+
+        private SsaIdentifier EnsureSsaIdentifier(Identifier id)
+        {
+            if (ssaCaller.Identifiers.TryGetValue(id, out var sid))
+                return sid;
+            sid = ssaCaller.Identifiers.Add(id, null, null, false);
+            var def = new DefInstruction(sid.Identifier);
+            var block = ssaCaller.Procedure.EntryBlock;
+            sid.DefStatement = block.Statements.Add(0, def);
+            return sid;
+        }
     }
```

Comment:
Lines 229-233 do the same as lines in the method `NewDefInstruction`1460-1464 in `SsaTransform.cs`. I suggest refactoring both, and creating a new method on `SsaState` called `MakeDefInstruction` which both can call.

Reply:
I have moved `EnsureSsaIdentifier` method from `SsaTransform` to `SsaState` in 3ffada84
The name `EnsureSsaIdentifier` is suggested by you (see https://github.com/uxmal/reko/pull/667#discussion_r209508554)
So I do not rename it to `MakeDefInstruction `